JIT: Assertion failed 'src1->OperIs(GT_LONG)'

Run `Windows_NT x86 Checked Build and Test (Jit - CoreFx)` CI tests will cause a failure on current master branch code:

```
20:10:24   Generated serialization assembly for assembly D:\j\workspace\x86_checked_w---92c0aac3\_\fx\bin\AnyOS.AnyCPU.Release\Microsoft.XmlSerializer.Generator.Tests\netstandard\Microsoft.XmlSerializer.Generator.Tests.dll --> 'D:\j\workspace\x86_checked_w---92c0aac3\_\fx\bin\AnyOS.AnyCPU.Release\Microsoft.XmlSerializer.Generator.Tests\netstandard\Microsoft.XmlSerializer.Generator.Tests.XmlSerializers.cs'.
20:10:25   Discovered:  Microsoft.Win32.Registry.Tests
20:10:32   Starting:    Microsoft.Win32.Registry.Tests
20:10:42   
20:10:42 
  Assert failure(PID 788 [0x00000314], Thread: 6912 [0x1b00]): Assertion failed 'src1->OperIs(GT_LONG)' in 'Microsoft.Xml.Serialization.GeneratedAssembly.XmlSerializationWriter1:Write65_LongEnum(long):ref:this' (IL size 90)
20:10:42   
20:10:42       File: d:\j\workspace\x86_checked_w---92c0aac3\src\jit\lower.cpp Line: 2167
20:10:42       Image: D:\j\workspace\x86_checked_w---92c0aac3\_\fx\bin\testhost\netcoreapp-Windows_NT-Release-x86\dotnet.exe
20:10:42   
20:10:43   Finished running tests.  End time=20:10:43.22, Exit code = 123456789
20:10:44 D:\j\workspace\x86_checked_w---92c0aac3\_\fx\Tools\tests.targets(445,5): warning MSB3073: The command "D:\j\workspace\x86_checked_w---92c0aac3\_\fx\bin/AnyOS.AnyCPU.Release/Microsoft.XmlSerializer.Generator.Tests/netstandard//RunTests.cmd D:\j\workspace\x86_checked_w---92c0aac3\_\fx\bin/testhost/netcoreapp-Windows_NT-Release-x86/" exited with code 123456789. [D:\j\workspace\x86_checked_w---92c0aac3\_\fx\src\Microsoft.XmlSerializer.Generator\tests\Microsoft.XmlSerializer.Generator.Tests.csproj]
20:10:44 
```

It should be introduced 3 days ago.

Please see the CI failure in this PR: https://github.com/dotnet/coreclr/pull/13480

--- a/src/jit/lower.cpp
+++ b/src/jit/lower.cpp
@@ -518,7 +518,7 @@ GenTree* Lowering::LowerSwitch(GenTree* node)
     assert(temp->gtOper == GT_LCL_VAR);
     unsigned   tempLclNum  = temp->gtLclVarCommon.gtLclNum;
     LclVarDsc* tempVarDsc  = comp->lvaTable + tempLclNum;
-    var_types  tempLclType = tempVarDsc->TypeGet();
+    var_types  tempLclType = temp->TypeGet();
 
     BasicBlock* defaultBB   = jumpTab[jumpCnt - 1];
     BasicBlock* followingBB = originalSwitchBB->bbNext;
@@ -548,7 +548,7 @@ GenTree* Lowering::LowerSwitch(GenTree* node)
     // both GT_SWITCH lowering code paths.
     // This condition is of the form: if (temp > jumpTableLength - 2){ goto jumpTable[jumpTableLength - 1]; }
     GenTreePtr gtDefaultCaseCond = comp->gtNewOperNode(GT_GT, TYP_INT, comp->gtNewLclvNode(tempLclNum, tempLclType),
-                                                       comp->gtNewIconNode(jumpCnt - 2, tempLclType));
+                                                       comp->gtNewIconNode(jumpCnt - 2, genActualType(tempLclType)));
 
     // Make sure we perform an unsigned comparison, just in case the switch index in 'temp'
     // is now less than zero 0 (that would also hit the default case).

--- a/tests/src/JIT/Regression/JitBlue/GitHub_13486/GitHub_13486.cs
+++ b/tests/src/JIT/Regression/JitBlue/GitHub_13486/GitHub_13486.cs
@@ -1,0 +1,33 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.CompilerServices;
+
+class Program
+{
+    enum LongEnum : long
+    {
+        Option0, Option1, Option2
+    }
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    static string Test(LongEnum v)
+    {
+        string s;
+        switch (v)
+        {
+        case LongEnum.Option0: s = "Option0"; break;
+        case LongEnum.Option1: s = "Option1"; break;
+        case LongEnum.Option2: s = "Option2"; break;
+        default: throw new Exception();
+        }
+        return s;
+    }
+
+    static int Main()
+    {
+        return (Test(LongEnum.Option0) == "Option0") ? 100 : 1;
+    }
+}

--- a/tests/src/JIT/Regression/JitBlue/GitHub_13486/GitHub_13486.csproj
+++ b/tests/src/JIT/Regression/JitBlue/GitHub_13486/GitHub_13486.csproj
@@ -1,0 +1,25 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
+  <PropertyGroup>
+    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
+    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
+    <AssemblyName>$(MSBuildProjectName)</AssemblyName>
+    <ProjectGuid>{95DFC527-4DC1-495E-97D7-E94EE1F7140D}</ProjectGuid>
+    <OutputType>Exe</OutputType>
+    <ProjectTypeGuids>{786C830F-07A1-408B-BD7F-6EE04809D6DB};{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}</ProjectTypeGuids>
+    <SolutionDir Condition="$(SolutionDir) == '' Or $(SolutionDir) == '*Undefined*'">..\..\</SolutionDir>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
+  </PropertyGroup>
+  <PropertyGroup>
+    <DebugType></DebugType>
+    <Optimize>False</Optimize>
+  </PropertyGroup>
+  <ItemGroup>
+    <Compile Include="$(MSBuildProjectName).cs" />
+  </ItemGroup>
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.targets))\dir.targets" />
+</Project>

